### PR TITLE
chat: add a mode specific class to the messages

### LINKED ISSUE #1807: Chat controls support different rendering mode like Text
### Description

It would be good to have chat controls support different rendring mode like Text. So it can handle markdown properly. 

### Solution Proposed

_No response_

### Impact of Solution

_No response_

### Additional Context

_No response_

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.
- [ ] Check if a new demo could be provided based on this, or if legacy demos could be benefit from it.
- [ ] Ensure any change is well documented.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
resolves #1807 
```
with tgb.Page() as page:
    tgb.toggle(theme=True)
    tgb.chat(
        "{conversation}",
        users=users,
        on_action=send_message,
        sender_id="Human",
        show_sender=True,
        height="80vh",
        mode="raw",
    )

```

### CODE REVIEW

**Review by dinhlongviolin1 — APPROVED** [review]

File: frontend/taipy-gui/src/components/Taipy/Chat.tsx [hunk 1]
```diff
@@ -156,7 +166,11 @@ const ChatRow = (props: ChatRowProps) => {
                         {!sender ? <Box sx={avatarColSx}>{avatar}</Box> : null}
                         <Stack>
                             <Box sx={sender ? rightNameSx : leftNameSx}>{name}</Box>
-                            <Paper sx={sender ? senderPaperSx : otherPaperSx} data-idx={index}>
+                            <Paper
+                                sx={sender ? senderPaperSx : otherPaperSx}
+                                data-idx={index}
+                                className={getSuffixedClassNames(className, "-" + mode)}
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/Chat.tsx:172]: I would have called these classes
    `taipy-chat-<mode>-[incoming|outgoing]-message`
and probably add
    `taipy-chat-[incoming|outgoing]-message` as well
and maybe add
    `taipy-chat-message`
    
    What say you?

**Review by FabienLelaquais — CHANGES_REQUESTED** [review]
Let's talk before I approve/reject

File: frontend/taipy-gui/src/components/Taipy/Chat.tsx (hunk 1 above)
Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Chat.tsx:172]: there's a class earlier in the hierarchy `taipy-chat-received` `taipy-chat-sent`

Comment (FredLL-Avaiga) [reply frontend/taipy-gui/src/components/Taipy/Chat.tsx:172]: is that enough for you ? @FabienLelaquais

Comment (FabienLelaquais) [reply frontend/taipy-gui/src/components/Taipy/Chat.tsx:172]: It is!

**Review by FabienLelaquais — APPROVED** [review]
👌

**Pull request merged**